src/test/java/teammates/test/cases/logic/CommentsLogicTest.java

```diff
@@ -370,6 +370,28 @@ public void testUpdateComment() throws Exception {
         assertEquals(1, actual.size());
         assertEquals(c.commentText, actual.get(0).commentText);
     }
+    
+    @Test
+    public void testUpdateCommentEmail() throws Exception {
+        CommentAttributes existingComment1 = dataBundle.comments.get("comment1FromI1C1toS1C1");
+        
+        ______TS("typical success case: email changed");
+        String courseId = existingComment1.courseId;
+        String oldInstrEmail = existingComment1.lastEditorEmail;
+        String updatedInstrEmail = "otheremail@gmail.tmt";
+        commentsLogic.updateInstructorEmail(courseId, oldInstrEmail, updatedInstrEmail);
+        
+        List<CommentAttributes> updatedComments = commentsLogic.getCommentsForGiver(courseId, updatedInstrEmail);
+        for (CommentAttributes updatedComment : updatedComments) {
+            assertEquals(updatedInstrEmail, updatedComment.giverEmail);
+            if (updatedComment.lastEditorEmail != null) {
+                assertEquals(updatedInstrEmail, updatedComment.lastEditorEmail);
+            }
+        }
+        
+        // edit back
+        commentsLogic.updateInstructorEmail(courseId, updatedInstrEmail, oldInstrEmail);
```

Comment:
Doesn't appear to cause any test failures with it removed, I guess it isn't necessary then.

Just to check, is the data bundle persisted at the start of every test in the class, or just before the first test in the class? I had initially assumed it was the latter, but I might be mistaken. The relevant section of code is as follows:

```java
    @BeforeClass
    public static void setupClass() throws Exception {
        printTestClassHeader();
        removeAndRestoreTypicalDataInDatastore();
    }
```